### PR TITLE
Add translator for Sud Ouest

### PR DESCRIPTION
**Pull request by symac**
Basic working version of a translator for the French newspaper sudouest.fr
Nothing much to add to it as there are not a lot of metadata available.

### CODE REVIEW

File: Sud Ouest.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	item.url = url;
+	item.abstract = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
+
+	// There might be a better option to isolate the author but this one works
+	var publishing = ZU.xpath(doc, "//div[@class='publishing']");
+	var author = Array.prototype.filter.call(publishing[0].childNodes, function (element) {
+		return element.nodeType === 3; /* Node.TEXT_NODE */
+	}).map(function (element) {
+		return element.textContent;
+	}).join("");
```
Comment (adam3smith) [inline Sud Ouest.js:105]: Why not just?
```suggestion
	var author = ZU.xpathText(doc, "//div[@class='publishing']/text()");
```

File: Sud Ouest.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+function scrape(doc, url) {
+	var item = new Zotero.Item("newspaperArticle");
+	item.title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	item.publication = "Sud Ouest";
+	item.ISSN = "1760-6454";
+	item.url = url;
+	item.abstract = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
+
+	// There might be a better option to isolate the author but this one works
+	var publishing = ZU.xpath(doc, "//div[@class='publishing']");
```
Comment (adam3smith) [inline Sud Ouest.js:100]: Maybe remove author when it's just generic institutional/agency?
https://www.sudouest.fr/2020/12/27/coronavirus-en-france-175-deces-en-24-heures-un-taux-de-positivite-toujours-en-baisse-8232175-10861.php

File: Sud Ouest.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("newspaperArticle");
+	item.title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	item.publication = "Sud Ouest";
+	item.ISSN = "1760-6454";
+	item.url = url;
+	item.abstract = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
+
```
Comment (adam3smith) [inline Sud Ouest.js:98]: add a snapshot

**Review by adam3smith — CHANGES_REQUESTED** [review]
Thanks! Some notes

**Comment by AbeJellinek** [comment]
@symac, any progress on these suggestions?

File: Sud Ouest.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
+{
+	"translatorID": "1f530308-ce67-4b87-a079-e4a8e2a9c623",
+	"label": "Sud Ouest",
+	"creator": "Sylvain Machefert",
+	"target": "https://www.sudouest.fr/",
```
[the remark this answers is not in the document]
Comment (symac) [reply Sud Ouest.js]: ok : 074b9bb0890817883b9e211805c5e29f08e41966

File: Sud Ouest.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+function detectWeb(doc, url) {
+	var publicationDate = getDateFromUrl(url);
+	
+	if (publicationDate) {
+		return 'newspaperArticle';
+	}
+	
+	if (ZU.xpath(doc, "//div[@class='articles-list']")) {
```
[the remark this answers is not in the document]
Comment (symac) [reply Sud Ouest.js]: Forgot it, yes it's better : 
https://github.com/zotero/translators/pull/2314/commits/65181321a36d6c4a33e560df6e76c87efea51159#diff-8347a42a7526460840a11937cd66ef51d40cd61c1e91857038062468237d80abR43

File: Sud Ouest.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+	
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('article.article .article-wrapper');
+	for (var i = 0; i < rows.length; i++) {
+		// Adjust if required, use Zotero.debug(rows) to check
```
[the remark this answers is not in the document]
Comment (symac) [reply Sud Ouest.js]: removed the comment

File: Sud Ouest.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+	item.abstract = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
+
+	// There might be a better option to isolate the author but this one works
+	var publishing = ZU.xpath(doc, "//div[@class='publishing']");
+	var author = Array.prototype.filter.call(publishing[0].childNodes, function (element) {
+		return element.nodeType === 3; /* Node.TEXT_NODE */
+	}).map(function (element) {
+		return element.textContent;
+	}).join("");
+
```
[the remark this answers is not in the document]
Comment (symac) [reply Sud Ouest.js:88]: yes, will always be fr-FR, set as default : https://github.com/zotero/translators/commit/65181321a36d6c4a33e560df6e76c87efea51159#diff-8347a42a7526460840a11937cd66ef51d40cd61c1e91857038062468237d80abR82

File: Sud Ouest.js [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+	// There might be a better option to isolate the author but this one works
+	var publishing = ZU.xpath(doc, "//div[@class='publishing']");
+	var author = Array.prototype.filter.call(publishing[0].childNodes, function (element) {
+		return element.nodeType === 3; /* Node.TEXT_NODE */
+	}).map(function (element) {
+		return element.textContent;
+	}).join("");
+	
+	author = author.replace("Par ", "");
+	item.creators.push({ lastName: author, fieldMode: 1, creatorType: "author" });
```
[the remark this answers is not in the document]
Comment (symac) [reply Sud Ouest.js]: it was copied/pasted from another translator, but this option seems to work better.

File: Sud Ouest.js (hunk 1 above)
Comment (symac) [reply Sud Ouest.js]: updated with this suggestion.

File: Sud Ouest.js (hunk 2 above)
Comment (symac) [reply Sud Ouest.js]: Testing for AFP/Sudouest which is often used. Might be improved later if it appears other generic names are used.

File: Sud Ouest.js (hunk 3 above)
Comment (symac) [reply Sud Ouest.js]: done

**Comment by symac** [comment]
Thanks @AbeJellinek for updating on this case. I have pushed a new version of the translator based on @adam3smith feedback. I have also updated the way to get the publication date as, meanwhile, sudouest.fr way of building url has changed.

Any new feedback welcome. Eslint works fine locally, but the CI fails, but not sure it is related to this specific PR.

**Comment by AbeJellinek** [comment]
Thanks! This looks great. CI only fails because it's having trouble finding the branch point. Nothing to do with your code. If you can update the tests (looks like just snapshots are missing), I'll go ahead and merge.

**Comment by symac** [comment]
you're right, I had forgotten to update the tests after adding the snapshot. Should now be fine.

**Comment by AbeJellinek** [comment]
Thanks!

**Pull request merged**